include/rrc/io/IOError.h

```diff
@@ -0,0 +1,26 @@
+/**
+ *  @autor nikitas
+ *  @date 10/13/16
+ */
+#pragma once
+
+#include <stdexcept>
+
+namespace rrc {
+
+    class IOError : public std::runtime_error {
+    public:
+        IOError();
+        IOError(const IOError &other);
+        IOError(bool error, const std::string &msg);
+
+        operator bool () const;
+
+    private:
+        bool mError;
```

Comment:
What does error attribute in your exception mean? When error triggered it could be not a error?